File: problems/0875-koko-eating-bananas/python/solution.py
def minEatingSpeed(piles, h):
    """Koko Eating Bananas — https://leetcode.com/problems/koko-eating-bananas/"""
    lo, hi = 1, max(piles)
    while lo < hi:
        mid = (lo + hi) // 2
        hours = sum((p + mid - 1) // mid for p in piles)
        if hours <= h:
            hi = mid
        else:
            lo = mid + 1
    return lo

File: problems/0875-koko-eating-bananas/typescript/solution.ts
// Koko Eating Bananas — https://leetcode.com/problems/koko-eating-bananas/
export function minEatingSpeed(piles: number[], h: number): number {
  let lo = 1;
  let hi = Math.max(...piles);
  while (lo < hi) {
    const mid = Math.floor((lo + hi) / 2);
    let hours = 0;
    for (const p of piles) hours += Math.ceil(p / mid);
    if (hours <= h) hi = mid;
    else lo = mid + 1;
  }
  return lo;
}
